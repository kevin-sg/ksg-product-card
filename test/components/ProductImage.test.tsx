import React from 'react';
import '../setupTest';
import { mount } from 'enzyme';

import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('Debe mostrar el componente con la imagen personalizado', () => {
    const wrapper = mount(<ProductImage img="https://hello-world.jpg" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar componente con la imagen del producto', () => {
    const wrapper = mount(<ProductCard product={product2}>{() => <ProductImage />}</ProductCard>);
    expect(wrapper).toMatchSnapshot();
  });
});
