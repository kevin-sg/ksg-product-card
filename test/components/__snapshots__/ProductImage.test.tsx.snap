// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostrar componente con la imagen del producto 1`] = `
<div
  className="productCard "
>
  <img
    alt="Product"
    className="productImg "
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe mostrar el componente con la imagen personalizado 1`] = `
<img
  alt="Product"
  className="productImg "
  src="https://hello-world.jpg"
/>
`;
